File: code_samples/csharp/async_promises.cs
using System;

/// <summary>
/// Async: Promises
/// AI/ML Training Sample
/// </summary>
public class Promises
{
    private string data;
    
    public Promises()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Promises();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}

File: code_samples/go/async_promises.go
package main

import (
    "fmt"
)

// Async: Promises
// AI/ML Training Sample

type Promises struct {
    Data string
}

func NewPromises() *Promises {
    return &Promises{
        Data: "",
    }
}

func (s *Promises) Process(input string) {
    s.Data = input
}

func (s *Promises) Validate() bool {
    return len(s.Data) > 0
}

func (s *Promises) GetData() string {
    return s.Data
}

func main() {
    instance := NewPromises()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
